src/main.cpp

```diff
@@ -7058,6 +7058,15 @@ bool ProcessMessage(CNode *pfrom, std::string strCommand, CDataStream &vRecv, in
         if (pfrom->ThinBlockCapable())
         {
             vRecv >> pfrom->nXthinBloomfilterSize;
+
+            // As a safeguard don't allow a smaller max bloom filter size than the default max size.
+            if (!pfrom->nXthinBloomfilterSize || (pfrom->nXthinBloomfilterSize < MAX_BLOOM_FILTER_SIZE))
```

Comment:
dumb question, why an spv client can't use a bloom filter smaller than MAX_BLOOM_FILTER_SIZE?

Reply:
I'm concerned with the disconnect choice here.  In this case it the MAX_BLOOM_FILTER_SIZE effectively becomes a consensus parameter because it could cause a netsplit if changed.  I see that when we change the param via bitcoin.conf this constant isn't changed.  However, given its more important role, I think that we need to move this to consensus.h and call it something like SMALLEST_MAX_BLOOM_FILTER_SIZE.

Reply:
ok, done